devicetypes/smartthings/ecobee-thermostat.src/ecobee-thermostat.groovy

```diff
@@ -135,16 +136,29 @@ metadata {
 void installed() {
     // The device refreshes every 5 minutes by default so if we miss 2 refreshes we can consider it offline
     // Using 12 minutes because in testing, device health team found that there could be "jitter"
-    sendEvent(name: "checkInterval", value: 60 * 12, data: [protocol: "cloud"], displayed: false)
+	initialize()
+}
+def initialize() {
+	sendEvent(name: "DeviceWatch-Enroll", value: JsonOutput.toJson([protocol: "cloud", scheme:"untracked"]), displayed: false)
+	updateDataValue("EnrolledUTDH", "true")
+}
+
+def updated() {
+	log.debug "updated()"
+	parent.setName(device.label, device.deviceNetworkId)
+	initialize()
+}
+
+// Called when the DTH is uninstalled, is this true for cirrus/gadfly integrations?
+// Informs parent to purge its associated data
+def uninstalled() {
+    log.debug "uninstalled() parent.purgeChildDevice($device.deviceNetworkId)"
+    // purge DTH from parent
+    parent?.purgeChildDevice(this)
 }
 
-// Device Watch will ping the device to proactively determine if the device has gone offline
-// If the device was online the last time we refreshed, trigger another refresh as part of the ping.
 def ping() {
```

Comment:
why is ping() needed for a C2C device?

Reply:
can you call updated() from ping() that way these devices can get enrolled properly. also, update the log statement to log.info "Ecobee ping() incorrectly called" or something. log.info will put it to sumo

Reply:
only time ping should be called is if the device is enrolled as tracked and there hasn't been any events sent from the thermostat for 12 minutes. As the thermostat now enrolls as untracked on the first poll response I don't se why we should add another call to updated when ping is called.